### ISSUE TITLE
Another pdfTeX text with pdfsizeopt.main.PdfXrefStreamError

### ISSUE THREAD

**Issue by rbrito** [pts/pdfsizeopt#36]
Dear @pts,

I just discovered another file that gives pdfsizeopt a hard time. Once again, it is a file that was produced by pdfTeX.

Here is what I get:

```
info: This is pdfsizeopt rUNKNOWN size=358996.
info: prepending to PATH: /home/rbrito/Downloads/pdfsizeopt
info: loading PDF from: PythonScientific-simple.pdf
info: loaded PDF of 16267367 bytes
info: found 7674 obj offsets and 34 obj streams in xref stream
Traceback (most recent call last):
  File "/home/rbrito/Downloads/pdfsizeopt/pdfsizeopt", line 37, in <module>
    sys.exit(main.main(sys.argv))
  File "/home/rbrito/Downloads/pdfsizeopt/lib/pdfsizeopt/main.py", line 8783, in main
    ).Load(file_name)
  File "/home/rbrito/Downloads/pdfsizeopt/lib/pdfsizeopt/main.py", line 3514, in Load
    do_ignore_generation_numbers=self.do_ignore_generation_numbers)
  File "/home/rbrito/Downloads/pdfsizeopt/lib/pdfsizeopt/main.py", line 3905, in ParseUsingXref
    xref_ofs, xref_obj_num, xref_generation)
  File "/home/rbrito/Downloads/pdfsizeopt/lib/pdfsizeopt/main.py", line 3848, in ParseUsingXrefStream
    compressed_obj_start))
pdfsizeopt.main.PdfXrefStreamError: location mismatch for objstm obj 175: objstm obj 168 has index 7, xref stream has (830, 63)
```

I'm uploading the file here.

[PythonScientific-simple.pdf](https://github.com/pts/pdfsizeopt/files/1265999/PythonScientific-simple.pdf)

**Comment by rbrito** [comment]
If I preprocess the file with qpdf before, the problem doesn't occur anymore. 😞

**Comment by pts** [comment]
Thank you for reporting this!

You can comment out the `raise` in line 3848, it's harmless. I'm preparing a proper fix.

**Comment by pts** [comment]
Fixed in 3c8d55cd2d669b390fd2c2744d9369a125995bb9.

The bug was: the obj_num checks in objstm objs were too strict in pdfsizeopt: even unused obj_nums were checked, and unused duplicate obj_nums were not allowed (but they should have been). The attached *PythonScientific-simple.pdf* contained unused duplicate obj_num 175.
